crates/sdk/src/network-v2/prover.rs

```diff
@@ -1,161 +1,199 @@
 use std::time::{Duration, Instant};
 
-use crate::network_v2::client::DEFAULT_PROVER_NETWORK_RPC;
-use crate::{
-    network_v2::client::NetworkClient,
-    network_v2::proto::network::{
-        ExecutionStatus, FulfillmentStatus, FulfillmentStrategy, ProofMode,
-    },
-    NetworkProverBuilder, Prover, SP1Context, SP1ProofKind, SP1ProofWithPublicValues,
-    SP1ProvingKey, SP1VerifyingKey,
-};
 use anyhow::Result;
 use backoff::{future::retry, Error as BackoffError, ExponentialBackoff};
 use serde::de::DeserializeOwned;
+use tokio::time::sleep;
+use tonic::Code;
+
 use sp1_core_machine::io::SP1Stdin;
 use sp1_prover::{components::DefaultProverComponents, SP1Prover, SP1_CIRCUIT_VERSION};
 use sp1_stark::SP1ProverOpts;
-use tonic::Code;
 
-use {crate::block_on, tokio::time::sleep};
+use crate::network_v2::{
+    client::{NetworkClient, DEFAULT_PROVER_NETWORK_RPC},
+    proto::network::{ExecutionStatus, FulfillmentStatus, FulfillmentStrategy, ProofMode},
+    types::{HashType, RequestId, VerifyingKeyHash},
+    Error,
+};
+use crate::{
+    block_on,
+    provers::{CpuProver, ProofOpts, ProverType},
+    NetworkProverBuilder, Prover, SP1Context, SP1ProofKind, SP1ProofWithPublicValues,
+    SP1ProvingKey, SP1VerifyingKey,
+};
+
+/// The default fulfillment strategy to use for proof requests.
+pub const DEFAULT_FULFILLMENT_STRATEGY: FulfillmentStrategy = FulfillmentStrategy::Hosted;
+
+/// The default timeout for a proof request to be fulfilled (4 hours).
+pub const DEFAULT_TIMEOUT_SECS: u64 = 14400;
 
-use crate::provers::{CpuProver, ProofOpts, ProverType};
+/// The minimum allowed timeout for a proof request to be fulfilled (10 seconds).
+pub const MIN_TIMEOUT_SECS: u64 = 10;
 
-/// The timeout for a proof request to be fulfilled.
-const TIMEOUT_SECS: u64 = 14400;
+/// The maximum allowed timeout for a proof request to be fulfilled (24 hours).
+pub const MAX_TIMEOUT_SECS: u64 = 86400;
 
-/// The default cycle limit for a proof request if simulation is skipped.
-const DEFAULT_CYCLE_LIMIT: u64 = 100_000_000;
+/// The default cycle limit for a proof request if simulation and the cycle limit is not explicitly
+/// set.
+pub const DEFAULT_CYCLE_LIMIT: u64 = 100_000_000;
 
 /// An implementation of [crate::ProverClient] that can generate proofs on a remote RPC server.
 pub struct NetworkProver {
     client: NetworkClient,
     local_prover: CpuProver,
-    skip_simulation: bool,
-    strategy: FulfillmentStrategy,
 }
 
 impl NetworkProver {
-    /// Creates a new [NetworkProver] with the given private key.
-    pub fn new(private_key: &str, rpc_url: Option<String>, skip_simulation: bool) -> Self {
+    /// Creates a new `NetworkProver` with the given private key.
+    pub fn new(private_key: &str) -> Self {
```

Comment:
nit: Should we really use a `string` here? Seems like `B256` may be a better choice.

Alternatively, Alloy has the same problem and has a reasonable type that they expose for this: https://github.com/alloy-rs/alloy/blob/main/crates/signer-local/src/lib.rs

Reply:
I'm open to it, although String seems the most versatile, as otherwise the client needs to import Alloy / whatever type we enforce.